function.go

```diff
@@ -67,29 +68,72 @@ func (f *Function) Initialize(runtime, context, name string, domainSearchLimit i
 
 	// Derive a name if not provided
 	if name == "" {
-		name = pathToDomain(f.root, domainSearchLimit)
+		name = pathToDomain(f.Root, domainSearchLimit)
 	}
 	if name == "" {
 		err = errors.New("Function name must be provided or be derivable from path")
 		return
 	}
-	f.name = name
+	f.Name = name
 
 	// Write the template implementation in the appropriate runtime
-	if err = initializer.Initialize(runtime, context, f.root); err != nil {
+	if err = initializer.Initialize(runtime, context, f.Root); err != nil {
 		return
 	}
 	// runtime was validated
-	f.runtime = runtime
+	f.Runtime = runtime
+
+	// An image tag for the function
+	f.Tag = tag
 
 	// Write out the state as a config file and return.
 	return writeConfig(f)
 }
 
+// WriteConfig writes this function's configuration to disk.
+func (f *Function) WriteConfig() (err error) {
+	return writeConfig(f)
+}
+
+// Initialized returns true if the function has been initialized with a name and a runtime
 func (f *Function) Initialized() bool {
 	// TODO: this should probably be more robust than checking what amounts to a
 	// side-effect of the initialization process.
-	return (f.runtime != "" && f.name != "")
+	return (f.Runtime != "" && f.Name != "")
+}
+
+// FunctionConfiguration accepts a path to a function project
+// and an image tag. It loads the existing function configuration
+// from disk at path. If the image tag parameter is not empty,
+// this will be used to update the function config file.
+// TODO: This should probably be changed to take a path
+// with optional overrides for both tag and name
+func FunctionConfiguration(path, tag string) (f *Function, err error) {
+	f, err = NewFunction(path)
+	if err != nil {
+		return nil, err
+	}
+
+	if f.Name == "" || f.Runtime == "" {
+		return nil, fmt.Errorf("Unable to find a function project at %s", path)
+	}
+
+	// Allow caller to override pre-configured t name
+	if tag != "" {
+		f.Tag = tag
+	}
+	if f.Tag == "" {
+		f.Tag = fmt.Sprintf("quay.io/%s:latest", f.Name)
```

Comment:
Possible future improvement: I think we could pick better default by looking into docker config file like 
 the `CompleteRegistryList` function does.